### PR TITLE
Reduce db load

### PR DESCRIPTION
**Pull request by doodyparizada**
we tried this on one machine on prod and it reduce the network from 10Mbit/sec to 100Kbit/sec to the DB.

### CODE REVIEW

File: scripts/src/models/applications.ts [hunk 1]
```diff
@@ -67,11 +56,13 @@ export class Application extends CreationDateModel {
...
 export class AppOffer extends BaseEntity {
 	public static async getAppOffers(appId: string, type: OfferType): Promise<AppOffer[]> {
-		// XXX add cache
 		const cacheKey = `appOffers:${appId}:${type}`;
-		// if (AppOffersCache.has(cacheKey)) {
-		// 	return AppOffersCache.get(cacheKey) as AppOffer[];
-		// }
+		if (AppOffersCache.has(cacheKey)) {
+			const [offers, lastReferesh] = AppOffersCache.get(cacheKey)!;
+			if (moment.duration(moment().diff(lastReferesh)).asMinutes() <= 10) {
```
Comment (liorama) [inline scripts/src/models/applications.ts]: Put the 10 into a const please

File: scripts/src/public/services/offer_contents.ts [hunk 2]
```diff
@@ -105,14 +107,15 @@ export async function getOfferContent(offerId: string): Promise<db.OfferContent
 }
 
 export async function getAllContents(): Promise<Map<string, db.OfferContent>> {
-	if (true) { // disable cache
+	if (moment.duration(moment().diff(lastReferesh)).asMinutes() > 10) {
```
Comment (liorama) [inline scripts/src/public/services/offer_contents.ts]: 10 into a const

**Comment by liorama** [comment]
Build failed

**Comment by doodyparizada** [comment]
@liorama , @Doc999tor is working on making this human readable

**Comment by nitzantomer** [comment]
what's up with these log files?

File: scripts/src/utils/cache.ts [hunk 3]
```diff
@@ -6,38 +6,27 @@ interface CacheValue {
 }
 
 interface CacheObj {
-	key: string;
-	value: CacheValue;
+	[key: string]: CacheValue;
```
Comment (nitzantomer) [inline scripts/src/utils/cache.ts]: you no longer need this `CacheObj`

File: scripts/src/models/applications.ts [hunk 4]
```diff
@@ -27,7 +17,7 @@ export type ApplicationConfig = {
 	limits: LimitConfig;
 };
 
-const AppOffersCache = new Map<string, AppOffer[]>();
+// const AppOffersCache = new Map<string, [AppOffer[], moment.Moment]>();
```
Comment (doodyparizada) [inline scripts/src/models/applications.ts]: delete comment

File: scripts/src/public/services/offer_contents.ts [hunk 5]
```diff
@@ -105,14 +105,17 @@ export async function getOfferContent(offerId: string): Promise<db.OfferContent
 }
 
 export async function getAllContents(): Promise<Map<string, db.OfferContent>> {
-	if (true) { // disable cache
-		const map = new Map<string, db.OfferContent>();
+	const cacheKey = "offerContents";
+	let contentsMap: Map<string, db.OfferContent> | null = localCache.get(cacheKey);
```
Comment (doodyparizada) [inline scripts/src/public/services/offer_contents.ts]: don't know which syntax is nicer, yours or this:
```typescript
let contentsMap = localCache.get<Map<string, db.OfferContent>>(cacheKey);
```

**Pull request merged**